api/v1alpha1/dynamicmodule_types.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+// DynamicModule defines a Dynamic Module extension.
+type DynamicModule struct {
+	// Name is a unique name for this Dynamic Module extension. It is used to identify the
+	// Dynamic Module extension if multiple extensions are loaded.
+	// It's also used for logging/debugging.
+	// If not specified, EG will generate a unique name for the Dynamic Module extension.
+	//
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// Module is the name of the dynamic module to load.
+	// The module name is used to search for the shared library file in the search path.
+	// The search path is configured by the environment variable ENVOY_DYNAMIC_MODULES_SEARCH_PATH.
+	// The actual search path is ${ENVOY_DYNAMIC_MODULES_SEARCH_PATH}/lib${name}.so.
+	//
+	// +kubebuilder:validation:Required
+	Module string `json:"module"`
+
+	// Config is the configuration for the Dynamic Module extension.
+	// This configuration will be passed to the Dynamic Module extension.
+	// +optional
+	Config *apiextensionsv1.JSON `json:"config,omitempty"`
```

Comment:
i am not sure if we want to force JSON all the time. The rationale behind why the Envoy API is Any is that we don't want to force dynamic modules to pull in dependencies to parse the input. Simply making this to string should work i guess?

Reply:
Related to the comment above, I would make this `ExtensionConfig` but not that strong preference

Reply:
Yep, make sense, updated